db/crud.go

```diff
@@ -48,10 +48,36 @@ func (db *DatabaseContext) GetDoc(docid string) (doc *document, err error) {
 	dbExpvars.Add("document_gets", 1)
 	if db.UseXattrs() {
 		var rawDoc, rawXattr []byte
-		cas, err := db.Bucket.GetWithXattr(key, KSyncXattrName, &rawDoc, &rawXattr)
-		doc, err = unmarshalDocumentWithXattr(docid, rawDoc, rawXattr, cas)
-		if err != nil {
-			return nil, err
+		importAttempts := 0
+		for {
+			cas, getErr := db.Bucket.GetWithXattr(key, KSyncXattrName, &rawDoc, &rawXattr)
+			if getErr != nil {
+				return nil, getErr
+			}
+			var unmarshalErr error
+			doc, unmarshalErr = unmarshalDocumentWithXattr(docid, rawDoc, rawXattr, cas)
+			if unmarshalErr != nil {
+				return nil, unmarshalErr
+			}
+			// If this was an SG Write, return the doc.  Otherwise, attempt to import
+			if doc.IsSGWrite() {
+				break
+			}
+
+			// Limit the number of import attempts for a given GET request, to guard against unexpected infinite looping
+			if importAttempts > 20 {
+				base.Warn("Unable to import non-SG write of document %s after 20 attempts", docid)
+				return nil, base.HTTPErrorf(404, "Not imported")
```

Comment:
Same comment -- I think the implications of this error should be concisely documented.  I wish it even pointed to a documentation page that explained it even further .. but I think a concise explanation would be a good start.